chapter07/chapter07.adoc

```diff
@@ -0,0 +1,276 @@
+
+*Prometheus* is a powerful tool designed to monitor and collect metrics from your services. It provides a highly efficient time-series database system that securely stores your data for long-term analysis. With Prometheus, you can easily visualize and gain insights into your system's performance, allowing you to make informed decisions and optimize your services for better efficiency and reliability.
+  
+=== Need for Metrics in Microservices
+
+Metrics, enables developers and operators to monitor and measure the behavior of microservices at runtime. This observability is crucial for:
+
+- *Performance Tuning*: Identifying bottlenecks and optimizing resource utilization to ensure services are running efficiently.
+- *Scalability*: Making informed decisions on when to scale services up or down based on real-time data on load and performance.
+- *Troubleshooting*: Quickly pinpointing issues by analyzing trends in performance metrics, leading to reduced downtime.
+- *Service Health Monitoring*: Complementing the MicroProfile Health checks by providing deeper insights into the internal state of a service, beyond simple up/down statuses.
+
+=== Types of Metrics
+
+MicroProfile Metrics offers a range of customizable metrics that can be used to measure and monitor microservices' performance. It allows developing microservices that are observable, manageable, and which provide insights into their behavior. 
+
+The MicroProfile Metrics specification includes four different types of metrics that serve specific monitoring purposes: Counter, Gauge, Histogram, and Timer. Each of these types offers unique insights into different aspects of application behavior and performance. Below is the breakdown of available metric types:
+
+1. *Counter*: A Counter is a simple metric type that represents a single numerical value that can only increase or be reset to zero. It's typically used to count occurrences of certain events, such as the number of requests processed, items created, or tasks completed. Since counters can only go up, they are resettable to start counting from zero again, which is particularly useful for tracking totals over a specific interval. 
+
+2. *Gauge*: It is a metric type that measures an instantaneous value of something , which can arbitrarily go up or down. It’s used to capture the value of a metric at a particular point in time like the size of a queue, memory usage, or current number of active user sessions. Gauges are typically used for values that fluctuate over time and provide a current "gauge" of what's happening.
+
+3. *Histograms*: They provide a distribution of values for a given metric, which are useful for identifying performance outliers. It measures the frequency of values in different ranges (or "buckets") and is useful for tracking the distribution of values, such as response times or data sizes. Histograms can give insights into the variability, average, median, percentiles, and trends of the measured data over time.
```

Comment:
In  5.x, standard deviation (variability) and median isn't provided anymore.

Reply:
Removed standard deviation (variability) and median.